src/core/namespace.c

```diff
@@ -1817,6 +1986,23 @@ int setup_namespace(
                 require_prefix = true;
         }
 
+        /* Create the source directory to allow runtime propagation of mounts.
+         * When working out ExtensionImages this will be needed already, just like
+         * the root directory is needed by other mounts, so set it up immediately. */
+        if (setup_propagate)
+                (void) mkdir_p(propagate_dir, 0600);
+
+        if (n_extension_images > 0) {
+                r = getenv_path_list("SYSTEMD_SYSEXT_HIERARCHIES", &hierarchies);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse SYSTEMD_SYSEXT_HIERARCHIES environment variable : %m");
+                if (!hierarchies) {
+                        hierarchies = strv_new("/usr", "/opt");
+                        if (!hierarchies)
+                                return -ENOMEM;
+                }
```

Comment:
maybe add a helper for this after all, to os-util.h or so, that gets the list of hierarchies.